app/src/main/java/com/babylon/wallet/android/presentation/transaction/submit/TransactionSubmitDelegate.kt

```diff
@@ -209,21 +210,42 @@ class TransactionSubmitDelegateImpl @Inject constructor(
             message = transactionRequest.unvalidatedManifestData.plainMessage,
             expiration = transactionRequestKind.expiration
         ).onSuccess { signedSubintent ->
-            _state.update { it.copy(isSubmitting = false) }
+            // Respond to dApp
+            respondToIncomingRequestUseCase.respondWithSuccessSubintent(
+                request = data.value.request,
+                signedSubintent = signedSubintent
+            )
+
+            appEventBus.sendEvent(
+                AppEvent.Status.PreAuthorization.Sent(
+                    requestId = transactionRequest.interactionId,
+                    preAuthorizationId = signedSubintent.subintent.hash(),
+                    isMobileConnect = transactionRequest.isMobileConnectRequest,
+                    dAppName = _state.value.proposingDApp?.name,
+                    remainingTime = when (val expiration = transactionRequestKind.expiration) {
+                        is SubintentExpiration.AtTime -> expiration.expirationDuration()
+                        is SubintentExpiration.DelayAfterSign -> expiration.delay
+                    }
+                )
+            )
 
-            // TODO poll subintent & send event
+            transactionStatusClient.startPollingForPreAuthorizationStatus(
+                intentHash = signedSubintent.subintent.hash(),
+                requestId = data.value.request.interactionId,
+                expiration = transactionRequestKind.expiration.toDAppInteraction()
+            )
 
-            // Respond to dApp
-            if (!data.value.request.isInternal) {
-                respondToIncomingRequestUseCase.respondWithSuccessSubintent(
-                    request = data.value.request,
-                    signedSubintent = signedSubintent
+            _state.update { it.copy(isSubmitting = false) }
+        }.onFailure { error ->
+            _state.update {
+                it.copy(
+                    isSubmitting = false,
+                    error = TransactionErrorMessage(error)
                 )
             }
-        }
+        }.toUnitResult()
     }
 
-    // TODO what about errors from sargon such as SubintentExpired.
```

Comment:
I didn't find any error cases (not covered already) when reaching this code in pre-auth case. I think we're good.